rEXIFier.sh

```diff
@@ -755,23 +705,63 @@ for WIPSortedFileAbsolutePath in ${WIPSortedFileAbsolutePaths[@]}; do
+                fi
+              # If the duplicate file exists, run deduplication process (the filename will be amended by adding an index)
               else
-                printf "Moving ${WIPSortedFileAbsolutePath} to ${DestinationPath}/${FileNameDuplicates}/${WIPDirectoryName}/${FormatedFileName} ${NOTE} SUCCESSFUL\n"
+                # Search for file basename match
+                MatchedFileNameDuplicates=$(find ${DestinationPath}/${FileNameDuplicates}/${WIPDirectoryName} -maxdepth 1 -type f -iname "${FormatedFileNameNoExtension}*.${NormalisedFileExtension}" )
+                # Sorting file names to process them sequentially
+                FileListSorter ${MatchedFileNameDuplicates}
+                # Returning the value from FileListSorter function by assigning the value of output (SortedFileList) to an array
+                # Bash functions, unlike functions in most programming languages do not allow you to return values to the caller
+                SortedMatchedFileNameDuplicates=${SortedFileList}
+                # Resetting SortedFileList array for sanity
+                SortedFileList=""
+                # Getting the last member in the sorted array of matched filenames, mind the difference in the construct 
+                # ${var[@]-1} finds the last element, versus ${var[@] -1} finds the last character of the last element
+                # ${var[*]-1} and ${var[*] -1} perform as above
+                LastSortedMatchedFileNameDuplicate=${SortedMatchedFileNameDuplicates[@]:-1}
+                # Extracting file basename
+                LastSortedMatchedFileDuplicateBasename=$(basename ${LastSortedMatchedFileNameDuplicate})
+                # Extracting file name
+                LastSortedMatchedFileNameDuplicate=${LastSortedMatchedFileDuplicateBasename%.*} 
+                # Extracting the duplicate index
+                LastSortedMatchedFileDuplicateIndex=$(echo ${LastSortedMatchedFileNameDuplicate} | awk -F "-" '{print $5}')
+                # Setting the value if the filename does not contain a duplicate index
+                if [[ LastSortedMatchedFileDuplicateIndex == "" ]] ; then
+                  LastSortedMatchedFileDuplicateIndex="00"
+                fi
+                # Incrementing the duplicate index so that we can form a new File Name
+                printf -v IncrementedLastSortedMatchedFileDuplicateIndex %02d "$((10#$LastSortedMatchedFileDuplicateIndex + 1))"
+                # Passing the incremented duplicate index to IncrementedDuplicateIndex
+                IncrementedDuplicateIndex=${IncrementedLastSortedMatchedFileDuplicateIndex}
+                # Re-forming the File Name with the incremented duplicate index
+                FormatedFileName=${FormatedFileNameNoExtension}-${IncrementedDuplicateIndex}.${NormalisedFileExtension}
```

Comment:
it feels like there is lot's of work just to find out the increment number, would be simpler just to add a random number at the end